### PR TITLE
Allow users to control level of Tracing spans

### PR DESCRIPTION
**Pull request by hannobraun**
## Motivation

I want to see the request method and path in my log message. See #123 for more information.

## Solution

This is a minimally invasive solution for #123: It allows the user to configure the level for request spans. While this is far from ideal API-wise, it's a quick fix. I don't think I know enough about how Tracing is supposed to be used to come up with something better right now (that spans have associated levels in the first place seems wrong to me, but I'm sure I'm missing something).

### CODE REVIEW

**Comment by davidbarsky** [comment]
(I'm putting on my "tracing contibutor" hat on.)

**Comment by davidbarsky** [comment]
(I'm putting on my "tracing contributor" hat on.)

Anyways! I noticed that in #123, you mentioned that you're creating the `TraceLayer`  thusly:

```rust
TraceLayer::new_for_http()
    .on_request(trace::DefaultOnRequest::new().level(Level::INFO))
    .on_response(trace::DefaultOnResponse::new().level(Level::INFO))
```

It seems to me that [`TraceLayer::make_span_with`](https://docs.rs/tower-http/0.1.1/tower_http/trace/struct.TraceLayer.html#method.new_for_http)—as documented [here](https://docs.rs/tower-http/0.1.1/tower_http/trace/index.html#recording-fields-on-the-span))—should address your use-case _without_ needing to introduce a new, slightly different API. Am I incorrect?

**Comment by hannobraun** [comment]
Oh, you're right! I was so focused on the `DefaultMakeSpan`, I somehow missed that you can pass your own `MakeSpan` implementation there. That addresses my use case.

Thank you!

**Comment by davidpdrsn** [comment]
I've opened. I still think it would be nice to address this inconsistency. Will review tomorrow.

File: tower-http/src/trace/on_request.rs [hunk 1]
```diff
@@ -60,10 +60,15 @@ impl DefaultOnRequest {
 
     /// Set the [`Level`] used for [tracing events].
     ///
+    /// Please note that while this will set the level for the tracing events
+    /// themselves, it might cause them to lack expected information, like
+    /// request method or path. You can address this using
+    /// [`DefaultMakeSpan::level`].
+    ///
```
Comment (davidpdrsn) [inline tower-http/src/trace/on_request.rs:67]: This bit feels redundant to me. The default span fields are already shown in the example output at the top of [the docs](https://docs.rs/tower-http/0.1.1/tower_http/trace/index.html). I kinda think we should remove this paragraph. Same goes for the change in `tower-http/src/trace/on_response.rs`.

**Review by davidpdrsn — CHANGES_REQUESTED** [review]
Sorry the delay!

I think this looks except for one small thing wrt the docs.

Also wanna add a line in the changelog?

You should also merge/rebase master to fix CI.

**Comment by hannobraun** [comment]
Thanks for the review and also sorry for the delay! I'm on vacation. I'll take care of this at some point during the next weeks.

**Comment by davidpdrsn** [comment]
No worries :)

File: tower-http/src/trace/on_request.rs (hunk 1 above)
Comment (hannobraun) [reply tower-http/src/trace/on_request.rs:67]: Hey @davidpdrsn, I don't understand what you mean here. It's true that the span fields are shown in the example output in the docs. But I was very confused when I used these `level` methods to set the log level for request/response, and then saw log messages that **did not** match this example output. I added the paragraph to address this point of confusion.

**Comment by hannobraun** [comment]
Rebased on top of master (no changes were necessary). Added changelog entry as requested. Did not address the review comment, as I disagree with the request (see comment there).

**Comment by davidpdrsn** [comment]
@hannobraun Sorry for the delay on this. I think it looks good as is 😊 Do you wanna merge/rebase master? I would like to include this in 0.1.2 which I'm hoping to release next week.

**Review by davidpdrsn — APPROVED** [review]

**Comment by davidpdrsn** [comment]
@hannobraun Thanks for working on this! Sorry for taking so long with reviews.

**Comment by hannobraun** [comment]
Looks like you got it sorted before I had a chance to look into it. Thanks!

And don't worry about the time it took. There are only so many hours in the day, after all :grin:

**Pull request merged**